cpp/arcticdb/processing/aggregation.cpp

```diff
@@ -489,4 +509,157 @@ SegmentInMemory CountAggregatorData::finalize(const ColumnName& output_column_na
     return res;
 }
 
+/***********************
+ * FirstAggregatorData *
+ ***********************/
+
+void FirstAggregatorData::add_data_type(DataType data_type) {
+    add_data_type_impl(data_type, data_type_);
+}
+
+void FirstAggregatorData::aggregate(const std::optional<ColumnWithStrings>& input_column, const std::vector<size_t>& groups, size_t unique_values) {
+    if(data_type_.has_value() && *data_type_ != DataType::EMPTYVAL && input_column.has_value()) {
+        entity::details::visit_type(*data_type_, [&input_column, unique_values, &groups, this] (auto global_type_desc_tag) {
+            using GlobalInputType = decltype(global_type_desc_tag);
+            using GlobalTypeDescriptorTag =  typename OutputType<GlobalInputType>::type;
+            using GlobalRawType = typename GlobalTypeDescriptorTag::DataTypeTag::raw_type;
+            aggregated_.resize(sizeof(GlobalRawType)* unique_values);
+            auto col_data = input_column->column_->data();
+            auto out_ptr = reinterpret_cast<GlobalRawType*>(aggregated_.data());
+            entity::details::visit_type(input_column->column_->type().data_type(), [this, &groups, &out_ptr, &col_data] (auto type_desc_tag) {
+                using ColumnTagType = std::decay_t<decltype(type_desc_tag)>;
+                using ColumnType =  typename ColumnTagType::raw_type;
+                auto groups_pos = 0;
+                auto grp_idx = groups[0];
```

Comment:
I'm confused by why we need this?

Reply:
I'm not sure I understand your question. `grp_idx` is used to check the current group we're in.

Reply:
Looking at the logic in lines 559-569, the behaviour would be the same if this variable didn't exist and it was removed from the if statements. Truth table:
```
grp_idx == groups[groups_pos] | is_first_group_el | Behaviour
true  | true  | groups_cache_.insert(groups[groups_pos]) and val = curr
true  | false | No action
false | true  | groups_cache_.insert(groups[groups_pos]) and val = curr
false | false | No action
```
i.e. the behaviour only depends on `is_first_group_el`, not on `grp_idx `

Reply:
Yes indeed! I guess leftovers when implementing, thanks!